test/framework/toy_build.py

```diff
@@ -2317,6 +2317,13 @@ def test_reproducibility_ext_easyblocks(self):
 
     def test_toy_toy(self):
         """Test building two easyconfigs in a single go, with one depending on the other."""
+
+        # skip when using Environment Modules tool, since this test generates an inconsistent environment
+        # (two modules with same name, expressing a reflexive conflict, one loading the other)
+        if isinstance(self.modtool, EnvironmentModules):
+            print("Skipping test_toy_toy because Environment Modules is being used")
+            return
```

Comment:
Rather than skipping this test, we should fix it: there's no explicit intention here to load two modules with the same name...

In `toy2.eb`, we could change the `name` field to `toy2`?

Reply:
I have changed the name field and other properties for `toy2.eb` to correctly build and install:

```python
"name = 'toy2'",
"easyblock = 'EB_toy'",
"sources = ['toy/toy-0.0.tar.gz']",
"patches = []",
"sanity_check_paths = { 'files': ['bin/toy2'], 'dirs': ['bin']}",
"prebuildopts = 'mv toy.source toy2.c &&'",
```